### PR TITLE
[JENKINS-47199] Use Jenkins Credential storage for JIRA credentials

### PR DESCRIPTION
**Pull request by nfalco79**
Use credential plugin as provider for JIRA credentials. Update the parent pom to latest version and use BOM for version management.

See [JENKINS-47199](https://issues.jenkins-ci.org/browse/JENKINS-47199).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description.
- [X] Appropriate unit or acceptance tests or explanation to why this change has no tests.
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [x] Run the changes and verified the change matches the issue description.
- [x] Reviewed the code.
- [x] Verified that the appropriate tests have been written or valid explanation given.
- [x] If applicable, tested by installing this plugin on the Jenkins instance.

### CODE REVIEW

**Comment by nfalco79** [comment]
I will add unit test is case the approach it's OK for mantainers

**Comment by nfalco79** [comment]
Test case added

**Comment by nfalco79** [comment]
@nrayapati the CI continue to fails with "Could not connect to EC2 (aws) - Ubuntu 18.04 LTS (i-0b810077b6108e84e) to send interrupt signal to process" that is not related to these changes. Do you have idea?

**Comment by nrayapati** [comment]
@nfalco79 Thank you very much for the PR!

**Comment by nrayapati** [comment]
Can you also add docs for this creds type at this page: https://jenkinsci.github.io/jira-steps-plugin/getting-started/config/authentication/

Source is at: https://github.com/jenkinsci/jira-steps-plugin/blob/master/hugo/content/getting-started/config/authentication.md

**Comment by nrayapati** [comment]
Verified by installed this snapshot on a local jenkins instance: 

Screen Shot 2020-05-16 at 1 31 19 AM
Screen Shot 2020-05-16 at 1 37 51 AM

**Comment by nrayapati** [comment]
> @nrayapati the CI continue to fails with "Could not connect to EC2 (aws) - Ubuntu 18.04 LTS (i-0b810077b6108e84e) to send interrupt signal to process" that is not related to these changes. Do you have idea?

Recently infra team switch these agents to be on EC2, seems like few environments issues. The build looks good now.

File: .gitignore [hunk 1]
```diff
@@ -25,3 +25,4 @@ _pdf
 *.settings
 *.bak
 public/
+/work
```
[the remark this answers is not in the document]
Comment (nfalco79) [reply .gitignore:28]: the `mvn hpi:run` creates a work folder in the root of the project with a jenkins instance used to develop and test

File: pom.xml [hunk 2]
```diff
@@ -84,16 +69,60 @@
...
+          <groupId>org.eclipse.jetty</groupId>
+          <artifactId>jetty-servlets</artifactId>
+          <version>9.4.28.v20200408</version>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+
   <dependencies>
+  <!-- limit dependency to only required
     <dependency>
```
[the remark this answers is not in the document]
Comment (nfalco79) [reply pom.xml:112]: ok, I will remove.

**Comment by nfalco79** [comment]
Updated

File: .gitignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (nrayapati) [reply .gitignore:28]: can you add the newline at the end for the file please

**Comment by nrayapati** [comment]
@nfalco79 Please don't force push it makes my life harder to review/verify everything again :)

**Comment by nfalco79** [comment]
ok, it was just to have a clean history. I had only change the pom.xml (workaggregator) and add the documented as you requested

**Comment by nrayapati** [comment]
> ok, it was just to have a clean history. I had only change the pom.xml (workaggregator) and add the documented as you requested

Regardless of how many commits we have on these PRS, I generally squash merge to keep the history clean. 

Thanks again for this PR, I will release this change soon.

**Pull request merged**